src/main/java/org/opensearch/security/privileges/SystemIndexAccessEvaluator.java

```diff
@@ -69,20 +70,28 @@ public class SecurityIndexAccessEvaluator {
     private final WildcardMatcher superAdminAccessOnlyIndexMatcher;
     private final WildcardMatcher deniedActionsMatcher;
 
+    private final Map<String, Set<String>> systemIndices;
     private final boolean isSystemIndexEnabled;
     private final boolean isSystemIndexPermissionEnabled;
 
-    public SecurityIndexAccessEvaluator(final Settings settings, AuditLog auditLog, IndexResolverReplacer irr) {
+    public SystemIndexAccessEvaluator(final Settings settings, AuditLog auditLog, IndexResolverReplacer irr, Map<String, Set<String>> systemIndices) {
         this.securityIndex = settings.get(
             ConfigConstants.SECURITY_CONFIG_INDEX_NAME,
             ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX
         );
         this.auditLog = auditLog;
         this.irr = irr;
+        this.systemIndices = systemIndices;
         this.filterSecurityIndex = settings.getAsBoolean(ConfigConstants.SECURITY_FILTER_SECURITYINDEX_FROM_ALL_REQUESTS, false);
-        this.systemIndexMatcher = WildcardMatcher.from(
-            settings.getAsList(ConfigConstants.SECURITY_SYSTEM_INDICES_KEY, ConfigConstants.SECURITY_SYSTEM_INDICES_DEFAULT)
+        List<String> allSystemIndices = settings.getAsList(
+                ConfigConstants.SECURITY_SYSTEM_INDICES_KEY,
+                ConfigConstants.SECURITY_SYSTEM_INDICES_DEFAULT
         );
+
+        if (systemIndices != null) {
+            allSystemIndices = systemIndices.values().stream().flatMap(Set::stream).collect(Collectors.toList());
+        }
+        this.systemIndexMatcher = WildcardMatcher.from(allSystemIndices);
```

Comment:
The idea here is to still fallback to the system index setting while adding a way for plugins to declare their system indices?

Reply:
What I would ultimately like to do is have the registry of system indices registered by plugins and modules passed to the security plugin from core and remove the `plugins.security.system_indices.indices` setting altogether.

The security plugin has notion of [ProtectedIndices](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/privileges/ProtectedIndexAccessEvaluator.java) which can be utilized to provide additional security to indices that are not system indices. 

The protected indices can be configured using:

```
plugins.security.protected_indices.enabled: true
plugins.security.protected_indices.indices: [index1, index2, index3, ...]
```

Reply:
Thanks for the details. However, won't this mean a plugin needs a code change + new release to register a system index v/s adding it to the setting currently? Also, do we have use-cases where system indices are non-plugin owned? or would protected indices be the suggested solution in such cases?

Reply:
In a few cases yes. See the comment on the linked issue here: https://github.com/opensearch-project/security/issues/4439#issuecomment-2163927300

There will be 2 categories of plugin updates with the 1st more common than the second.

1. **Plugin exists in plugins.security.system_indices.indices but is not registered via SystemIndexPlugin.getSystemIndexDescriptor** - These will be small updates to plugins like ml-commons that have added `.plugins-ml*` to `plugins.security.system_indices.indices`, but have not implemented SystemIndexPlugin.getSystemIndexDescriptor
2. Plugins that have implemented SystemIndexPlugin.getSystemIndexDescriptor, but have not added to `plugins.security.system_indices.indices`.
   - This is the case with job-scheduler and I have opened up a PR to ensure interactions with the system index are wrapped in threadContext.stashContext: https://github.com/opensearch-project/job-scheduler/pull/347
   - This is also the case with [`opensearch-dashboards` module](https://github.com/opensearch-project/OpenSearch/tree/main/modules/opensearch-dashboards). I've been analyzing that module in more detail and it looks like its currently unused. I will open up an issue on the core repo to discuss further.

Reply:
> Also, do we have use-cases where system indices are non-plugin owned?

All system indices are either owned by modules or plugins

Reply:
Opened an issue in core to discuss the `opensearch-dashboards` module in greater detail to understand its purpose (or lack thereof): https://github.com/opensearch-project/OpenSearch/issues/14500